django-stubs/db/models/fields/related_descriptors.pyi

```diff
@@ -36,19 +38,31 @@ class ForwardManyToOneDescriptor(Generic[_F]):
 class ForwardOneToOneDescriptor(ForwardManyToOneDescriptor[_F]):
     def get_object(self, instance: Model) -> Model: ...
 
-class ReverseOneToOneDescriptor:
+class ReverseOneToOneDescriptor(Generic[_From, _To]):
+    """
+    In the example::
+
+        class Restaurant(Model):
+            place = OneToOneField(Place, related_name='restaurant')
+
+    ``Place.restaurant`` is a ``ReverseOneToOneDescriptor`` instance.
+    """
+
     related: OneToOneRel
     def __init__(self, related: OneToOneRel) -> None: ...
     @property
     def RelatedObjectDoesNotExist(self) -> type[ObjectDoesNotExist]: ...
-    def is_cached(self, instance: Model) -> bool: ...
-    def get_queryset(self, **hints: Any) -> QuerySet: ...
+    def is_cached(self, instance: _From) -> bool: ...
+    def get_queryset(self, **hints: Any) -> QuerySet[_To]: ...
```

Comment:
The plugin should probably refine `def get_queryset()`s return type to be from the default manager of the related model. Can be seen here:

https://github.com/django/django/blob/574ee4023e15cfb195833edfbaed353f8021c62f/django/db/models/fields/related_descriptors.py#L443-L444

I wanted to keep the change set smaller, so we can do this in a subsequent PR. As we already do a bit of changes and everything in here should be improvements. In general I think the usage is quite small for `get_queryset` in here anyways..